staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_test.go

```diff
@@ -30,3 +33,43 @@ func TestNilUnstructuredContent(t *testing.T) {
 	assert.EqualValues(t, expContent, content)
 	assert.Equal(t, uCopy, &u)
 }
+
+// TestUnstructuredMetadataOmitempty checks that ObjectMeta omitempty
+// semantics are enforced for unstructured objects.
+// Example: the metadata.clusterName field has the omitempty json tag
+// so if it is set to it's zero value (""), it should be removed from the metadata map.
+func TestUnstructuredMetadataOmitempty(t *testing.T) {
+	var u Unstructured
+
+	// start with an empty metadata
+	emptyMetadata := make(map[string]interface{})
+	u.setNestedField(emptyMetadata, "metadata")
+
+	// set zero values for all fields in metadata explicitly
+	// since all fields in objectMeta are omitempty, these fields should never be set
+	// and the resulting metadata should be an empty metadata
+	u.SetName("")
```

Comment:
shouldn't we set random values first?

Reply:
I meant to do fuzzing before these to catch missing calls. If a field is added to ObjectMeta now, this list gets incomplete, but we don't notice. If we fuzz before, we would.